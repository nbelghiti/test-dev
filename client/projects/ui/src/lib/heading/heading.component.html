<div
	class="c-heading {{ classes }}"
	[ngClass]="{'c-heading--inline': inline}"
>
	<ng-container [ngSwitch]="level">
		<h1 *ngSwitchCase="HeadingLevel.heading1">{{title}}</h1>
		<h2 *ngSwitchCase="HeadingLevel.heading2">{{title}}</h2>
		<h3 *ngSwitchCase="HeadingLevel.heading3">{{title}}</h3>
	</ng-container>

	<div class="c-heading__meta">
		<i
			*ngIf="loading; else withContent"
			class="fas fa-{{IconName.spinner}} fa-spin c-heading__loader"
		>
		</i>

		<ng-template #withContent>
			<ng-content></ng-content>
		</ng-template>
	</div>
</div>
