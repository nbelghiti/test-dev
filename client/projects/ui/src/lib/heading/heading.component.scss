// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Heading
// =============================================================================
.c-heading {
	display: inline-flex;
	flex-direction: column;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-heading__loader {
	color: $primary-color;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-heading--inline {
	flex-direction: row;
	align-items: baseline;

	.c-heading__meta {
		margin-top: 0;
		margin-left: 1rem;
	}
}
