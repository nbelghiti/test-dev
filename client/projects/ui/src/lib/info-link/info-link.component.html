<a
	#button
	[routerLink]="link"
	class="c-info-link"
	[ngClass]="{
		'c-info-link--background-grey': grey,
		'c-info-link--background-red': red
	}"
>
	<div class="c-info-link__content">
		<span class="c-info-link__icon">
			<ui-icon [name]="icon" size="large"></ui-icon>
		</span>

		<ui-heading
			class="c-info-link__title"
			level="3"
			[title]="title"
		></ui-heading>
	</div>

	<span
		*ngIf="arrow"
		class="c-info-link__arrow"
	>
		<ui-icon name="arrow-right"></ui-icon>
	</span>
</a>
