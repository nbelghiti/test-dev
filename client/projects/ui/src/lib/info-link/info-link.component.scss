// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Info Link
// =============================================================================
.c-info-link {
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	padding: 1.5rem 3rem;

	text-decoration: none;
	text-align: left;

	border-bottom: $border-width-base solid $border-color-grey-76;

	cursor: pointer;

	&:hover {

		.c-info-link__arrow {
			transform: translateX(0.8rem);
		}
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-info-link__content {
	display: flex;
	align-items: center;
}

.c-info-link__icon {
	display: flex;
	justify-content: center;
	width: 2.4rem;
	margin-right: 1.6rem;

	line-height: $line-height-normalize;

	transition: color $animation-base;
}

.c-info-link__arrow {
	display: flex;

	transition: transform $animation-base;
}

.c-info-link__title {
	transition: color $animation-base;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-info-link--background-grey {
	background-color: $background-color-grey-96;
}

.c-info-link--background-red {
	background-color: $ui-color-error-light;
}
