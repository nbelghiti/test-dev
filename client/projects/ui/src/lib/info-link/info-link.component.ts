import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

@Component({
	selector: 'ui-info-link',
	templateUrl: './info-link.component.html',
	styleUrls: ['./info-link.component.scss']
})

export class InfoLinkComponent {
	@Input() public icon: IconName;
	@Input() public title: string;
	@Input() public link?: string;
	@Input() public grey?: boolean = false;
	@Input() public red?: boolean = false;
	@Input() public arrow?: boolean = true;
}
