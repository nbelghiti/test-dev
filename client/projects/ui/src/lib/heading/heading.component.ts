import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { HeadingLevel } from './heading.types';

@Component({
	selector: 'ui-heading',
	templateUrl: './heading.component.html',
	styleUrls: ['./heading.component.scss']
})

export class HeadingComponent {
	@Input() public title: string;
	@Input() public classes: string = '';
	@Input() public inline: boolean = false;
	@Input() public loading: boolean = false;
	@Input() public level: HeadingLevel = HeadingLevel.heading1;

	public HeadingLevel: typeof HeadingLevel = HeadingLevel;
	public IconName: typeof IconName = IconName;
}
