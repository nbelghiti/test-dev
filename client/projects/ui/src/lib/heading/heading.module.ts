import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingComponent } from './heading.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		HeadingComponent,
	],
	exports: [
		HeadingComponent,
	],
})
export class HeadingModule {}
