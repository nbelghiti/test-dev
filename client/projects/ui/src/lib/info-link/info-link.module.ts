import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { HeadingModule } from '../heading/heading.module';
import { IconModule } from '../icon/icon.module';

import { InfoLinkComponent } from './info-link.component';

@NgModule({
	imports: [
		CommonModule,
		RouterModule,
		IconModule,
		HeadingModule,
	],
	declarations: [
		InfoLinkComponent,
	],
	exports: [
		InfoLinkComponent,
	],
})
export class InfoLinkModule {}
