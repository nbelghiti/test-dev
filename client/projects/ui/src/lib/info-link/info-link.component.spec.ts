import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { HeadingComponent } from '../heading/heading.component';
import { IconComponent } from '../icon/icon.component';

import { InfoLinkComponent } from './info-link.component';

describe('InfoLinkComponent', () => {
	let component: InfoLinkComponent;
	let fixture: ComponentFixture<InfoLinkComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
			],
			declarations: [
				InfoLinkComponent,
				IconComponent,
				HeadingComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(InfoLinkComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

