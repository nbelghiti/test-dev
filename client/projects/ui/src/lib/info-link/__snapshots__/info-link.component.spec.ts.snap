// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLinkComponent Should render correctly 1`] = `
<ui-info-link
  arrow={[Function Boolean]}
  grey="false"
  red="false"
>
  <a
    class="c-info-link"
    href="/"
    ng-reflect-klass="c-info-link"
    ng-reflect-ng-class="[object Object]"
  >
    <div
      class="c-info-link__content"
    >
      <span
        class="c-info-link__icon"
      >
        <ui-icon
          class="c-icon"
          ng-reflect-size="large"
          size="large"
        >
          <i
            class="c-icon__glyph c-icon__glyph--large  fas fa-"
          />
        </ui-icon>
      </span>
      <ui-heading
        class="c-info-link__title"
        level="3"
        ng-reflect-level="3"
      >
        <div
          class="c-heading"
          ng-reflect-klass="c-heading "
          ng-reflect-ng-class="[object Object]"
        >
          
          
          
          
          <h3>
            
          </h3>
          <div
            class="c-heading__meta"
          >
            
            
            
          </div>
        </div>
      </ui-heading>
    </div>
    
    <span
      class="c-info-link__arrow"
    >
      <ui-icon
        class="c-icon"
        name="arrow-right"
        ng-reflect-name="arrow-right"
      >
        <i
          class="c-icon__glyph   fas fa-arrow-right"
        />
      </ui-icon>
    </span>
  </a>
</ui-info-link>
`;
